packages/block-library/src/search/edit.native.js

```diff
@@ -154,112 +155,165 @@ export default function SearchEdit( {
-				}
-				onFocus={ () => {
-					setIsPlaceholderSelected( true );
-					onFocus();
-				} }
-				onBlur={ () => setIsPlaceholderSelected( false ) }
-			/>
+			<View
+				style={ styles.searchInputContainer }
+				accessible={ true }
+				accessibilityRole="none"
+				accessibilityHint={ __(
+					'Double tap to edit placeholder text'
+				) }
+				accessibilityLabel={ `${ __(
+					'Search input field. Current placeholder value is'
+				) } ${ placeholder }` }
+			>
+				<PlainText
+					ref={ textInputRef }
+					isSelected={ isPlaceholderSelected }
+					className="wp-block-search__input"
+					style={ inputStyle }
+					numberOfLines={ 1 }
+					ellipsizeMode="tail" // currently only works on ios
+					label={ null }
+					value={ placeholder }
+					placeholder={
+						placeholder ? undefined : __( 'Optional placeholder…' )
+					}
+					onChange={ ( newVal ) =>
+						setAttributes( { placeholder: newVal } )
+					}
+					onFocus={ () => {
+						setIsPlaceholderSelected( true );
+						onFocus();
+					} }
+					onBlur={ () => setIsPlaceholderSelected( false ) }
+					placeholderTextColor={ placeholderStyle.color }
+				/>
+			</View>
 		);
 	};
 
+	// To achieve proper expanding and shrinking `RichText` on Android, there is a need to set
+	// a `placeholder` as an empty string when `RichText` is focused,
+	// because `AztecView` is calculating a `minWidth` based on placeholder text.
+	const buttonPlaceholderText =
+		isButtonSelected ||
+		( ! isButtonSelected && buttonText && buttonText !== '' )
+			? ''
+			: __( 'Add button text' );
+
 	const renderButton = () => {
 		return (
 			<View style={ styles.buttonContainer }>
-				{ buttonUseIcon && (
-					<Button
-						className="wp-block-search__button"
-						icon={ search }
-						onFocus={ onFocus }
-					/>
-				) }
+				{ buttonUseIcon && <Icon icon={ search } { ...styles.icon } /> }
 
 				{ ! buttonUseIcon && (
+					<View
+						accessible={ true }
+						accessibilityRole="none"
+						accessibilityHint={ __(
+							'Double tap to edit button text'
+						) }
+						accessibilityLabel={ `${ __(
+							'Search button. Current value is'
+						) } ${ buttonText }` }
+					>
+						<RichText
+							className="wp-block-search__button"
+							identifier="text"
+							tagName="p"
+							style={ styles.richTextButton }
+							placeholder={ buttonPlaceholderText }
+							value={ buttonText }
+							withoutInteractiveFormatting
+							onChange={ ( html ) =>
+								setAttributes( { buttonText: html } )
+							}
+							minWidth={ MIN_BUTTON_WIDTH }
+							textAlign="center"
+							isSelected={ isButtonSelected }
+							__unstableMobileNoFocusOnMount={ ! isSelected }
+							unstableOnFocus={ () => {
+								setIsButtonSelected( true );
+							} }
+							onBlur={ () => {
+								setIsButtonSelected( false );
+							} }
+							selectionColor={ styles.richTextButtonCursor.color }
+						/>
+					</View>
+				) }
+			</View>
+		);
+	};
+
+	return (
+		<View
```

Comment:
I found moving focus to this element on iOS announces "Search block, row 53, button, **actions available**." I'm unsure why "actions available" is announced or where it originates. Do you have any insight into this?

Reply:
I spent quit a bit of time trying to figure this one out and I believe it's due to the use of `<PlainText/>`. The `Code` block using this component as well and it also announces "actions available". The `File` block uses it too, but the `PlainText` component has the `editable={false}` property set so it doesn't announce "actions available". If I do the same thing on my block that announcement also goes away so I'm guessing this is VoiceOver recognizing that a native editable component is a child of the Search block.